lisa/datautils.py

```diff
@@ -504,6 +504,40 @@ def df_window(df, window, method='inclusive', clip_window=False):
     """
     return _data_window(df, window, method, clip_window)
 
+def _data_adjacent_index(data, ref, prev):
+    """
+    ``data`` can either be a :class:`pandas.DataFrame` or :class:`pandas.Series`.
+    """
+    res = None
+
+    if prev and ref > data.index[0]:
+        res = data.index.get_loc(ref, "ffill")
+    elif ref < data.index[-1]:
+        res = data.index.get_loc(ref, "bfill")
+
+    return res
+
+def df_adjacent_index(df, ref, prev=True):
+    """
+    :param ref: The index value to look around
+
+    :param prev: Whether to find the previous index (``True``)
+      or the next index (``False``)
+    :type prev: bool
+
+    :returns: The index value directly preceding ``ref`` if ``prev``,
```

Comment:
That's not strictly the case.: if df[ref] exists, it will be returned instead of the prev/next value.

Maybe something like:
```
The row at index directly preceding ``ref`` if ``prev`` if (``ref`` is not present in the index)
```